### PR TITLE
Frontend feedback form

### PR DESCRIPTION
**Pull request by ALI-KHALED-13**

### CODE REVIEW

**Comment by ALI-KHALED-13** [comment]
test comment

**Comment by ALI-KHALED-13** [comment]
test

File: client/src/pages/FeedbackForm/index.tsx [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+          <Button
+            onClick={()=> navigate(-1)}
+            color="navy"
+            hoverColor="lightNavy"
+            style={{marginLeft: "auto"}}
+            type="button"
+          >
+            Cancel
+          </Button>
+          <Button color="violet" /* should it be product theme color? */ hoverColor="lightViolet">
```
Comment (Elena-Erganyan) [inline client/src/pages/FeedbackForm/index.tsx:131]: I would say let's keep it simple, i.e. violet everywhere

File: client/src/pages/FeedbackForm/index.tsx [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+import Button from "../../components/Button";
+import { StyledButtonsWrapper } from "./styled";
+import { useGetTagsOptions, useStatusData } from "./utils";
+import { useAddFeedbackMutation } from "../../redux/api/feedbackApi";
+import { getErrorMessage } from "../../utils/getErrorMessage";
+
+
+
+const FeedbackForm =()=> {
```
Comment (Elena-Erganyan) [inline client/src/pages/FeedbackForm/index.tsx:19]: I understand it's half baked for now (I can see many unused variables/imports). So I will check it thoroughly once you are done with editing and deleting as well.

File: client/src/pages/FeedbackForm/index.tsx [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+      />
+      <DropDown
+        label="Category - tag"
+        options={tagsOptions}
+        value={tags[0]}
+        onChange={(selectedOp)=> setTags([selectedOp])}
+      />
+      <DropDown
+        label="Update Status"
+        note="change feedback status"
```
Comment (Elena-Erganyan) [inline client/src/pages/FeedbackForm/index.tsx:105]: Should look consistent. The note and the status should be capitalized
Screenshot_1

**Review by Elena-Erganyan — CHANGES_REQUESTED** [review]
The addition works, well done 😃 
Left some comments
More of review will be done once you finish with the rest part

File: models/Product.js [hunk 4]
```diff
@@ -21,14 +21,26 @@ const productSchema = new mongoose.Schema({
...
         return true
       }
     }
   },
+  tags: {
+    type: [String],
+    required: true,
+    validate: {
+      validator(tagsArr){
+        if (tagsArr.length < 3){
```
[the remark this answers is not in the document]
Comment (ALI-KHALED-13) [reply models/Product.js:37]: 8?

File: client/src/pages/ProductForm/index.tsx [hunk 5]
```diff
@@ -159,6 +163,14 @@ const ProductForm =()=> {
         required={false}
         onChange={(ev)=> setLink(ev.target.value)}
       />
+      <TagInput
```
[the remark this answers is not in the document]
Comment (ALI-KHALED-13) [reply client/src/pages/ProductForm/index.tsx:166]: fixed.. thanks

File: client/src/pages/Dashboard/index.tsx [hunk 6]
```diff
@@ -59,7 +64,12 @@ const Dashboard =()=> {
     <MetaHelmet
       title="Feedback Hub - Dashboard"
       canonical="/dashboard"
-    />    
+    />
+    <AlertMessage
```
Comment (Elena-Erganyan) [inline client/src/pages/Dashboard/index.tsx:68]: Just a note: when I went through the link in the dashboard and deleted the feedback it redirected me to the product page, not to this one 🤔

**Review by Elena-Erganyan — CHANGES_REQUESTED** [review]
1) Logged in as @hoho, created a feedback on the `Feedback Hub` product. @hoho is neither an owner nor a contributor of a product. In theory @hoho should be able to edit his own feedback (except the status). However, when I tried to edit some info and save it the following error occurred:
Screenshot_1
2) Logged in as myself. I'm a contributor of the `Feedback Hub` product. I should be able to change only a status of any feedback. When I tried to do that and save the feedback I had this error
Screenshot_2
3) Logged in as @hoho and added a feedback to the `Test product` with the status "Planned" (I could do because @hoho is the owner of the product). When I tried to change the status it said:
Screenshot_3

4) In all those cases if I try to edit and save again it's not saved and I don't see any error alert anymore.
5) When I deleted a feedback and refreshed the product page the alert message that the feedback was deleted still appeared (probably because of some info in history). Not a critical bug, low priority, can be ignored for now 😝

**Comment by ALI-KHALED-13** [comment]
@Elena-Erganyan solved bugs number 1, 2 and hopefully 3 (forgot to test the later XD)
issue 4 is related to alertMessage again ;( .. I must look for a definite solution
issue 5 will be investigated too

**Review by Elena-Erganyan — APPROVED** [review]
Good job 👏

**Pull request merged**